### PR TITLE
fix: fees part1 cleanup

### PR DESCRIPTION
**Pull request by elmutt**
- Use get feeData instead of buildSendTransaction to estimate fees.
- Make it work with new btc specific fee data from lib

- Depends on this pr: https://github.com/shapeshift/lib/pull/190

https://github.com/shapeshift/web/issues/215

This is part 1 of a series of PR's to clean up how to handle fees

### CODE REVIEW

File: src/components/Modals/Send/hooks/useFormSend/useFormSend.tsx [hunk 1]
```diff
@@ -1,5 +1,7 @@
 import { useToast } from '@chakra-ui/react'
-import { utxoAccountParams } from '@shapeshiftoss/chain-adapters'
+import {utxoAccountParams } from '@shapeshiftoss/chain-adapters'
+import { ChainTypes } from '@shapeshiftoss/types'
+import { FeeData } from '@shapeshiftoss/types/dist/chain-adapters'
```
Comment (cjthompson) [inline src/components/Modals/Send/hooks/useFormSend/useFormSend.tsx:4]: we should not be importing from `dist` directly

File: src/components/Modals/Send/hooks/useFormSend/useFormSend.tsx [hunk 2]
```diff
@@ -24,27 +26,50 @@ export const useFormSend = () => {
...
+          const accountParams = accountType ? utxoAccountParams(data.asset, accountType, 0) : {}
+          result = await adapter.buildSendTransaction({
+            to: data.address,
+            value,
+            erc20ContractAddress: data.asset.tokenId,
+            wallet,
+            fee,
+            gasLimit,
+            ...accountParams
+          })
```
Comment (cjthompson) [inline src/components/Modals/Send/hooks/useFormSend/useFormSend.tsx:68]: Is there any common logic/data structures here that can be consolidated and minimize the `if (chain)` code.

File: src/components/Modals/Send/hooks/useFormSend/useFormSend.tsx (hunk 1 above)
Comment (elmutt) [reply src/components/Modals/Send/hooks/useFormSend/useFormSend.tsx:4]: good catch

File: src/components/Modals/Send/hooks/useFormSend/useFormSend.tsx (hunk 2 above)
Comment (elmutt) [reply src/components/Modals/Send/hooks/useFormSend/useFormSend.tsx:68]: ya there is. good call

Comment (elmutt) [reply src/components/Modals/Send/hooks/useFormSend/useFormSend.tsx]: done

File: src/components/Modals/Send/hooks/useFormSend/useFormSend.tsx (hunk 1 above)
Comment (elmutt) [reply src/components/Modals/Send/hooks/useFormSend/useFormSend.tsx]: done

**Review by cjthompson — APPROVED** [review]

File: src/components/Modals/Send/hooks/useFormSend/useFormSend.tsx [hunk 3]
```diff
@@ -51,19 +51,14 @@ export const useFormSend = () => {
...
-          const fees = estimatedFees[feeType] as FeeData<ChainTypes.Bitcoin>
+          const fees = estimatedFees[feeType] as chainAdapters.FeeData<ChainTypes.Bitcoin>
           const fee = fees.feePerUnit
-          const gasLimit = fees.chainSpecific?.byteCount
-          const accountType = allAccountTypes[getAccountTypeKey(data.asset.chain)]
-          const accountParams = accountType ? utxoAccountParams(data.asset, accountType, 0) : {}
           result = await adapter.buildSendTransaction({
             to: data.address,
             value,
             erc20ContractAddress: data.asset.tokenId,
```
Comment (0xdef1cafe) [inline src/components/Modals/Send/hooks/useFormSend/useFormSend.tsx:59]: remove this too please, bitcorn don't have no contracts

**Review by 0xdef1cafe — APPROVED** [review]

**Review by DaoDev44 — APPROVED** [review]

**Pull request merged**